src/tesh/tests/fixtures/whitespace.md

```diff
@@ -0,0 +1,61 @@
+# Whitespace handling in tesh code blocks
+
+# Leading whitespace in command output is matched
+
+## Code block flush with left margin
+
+```console tesh-session="non-indented-block"
+$ # No blank lines after these commands:
+$ printf '0 space\n  2 space\n    4 space\n'
+0 space
+  2 space
+    4 space
+$ printf '  2 space\n    4 space\n0 space\n'
+  2 space
+    4 space
+0 space
+$ # Blank lines after these examples:
+$ printf '0 space\n  2 space\n    4 space\n'
+0 space
+  2 space
+    4 space
+
+
+$ printf '  2 space\n    4 space\n0 space\n'
+  2 space
+    4 space
+0 space
+
+
```

Comment:
One newline too many?

Reply:
Yeah, I think so!

Reply:
For some more context on this, I've found that the fact that tesh ignores trailing whitespace means you can leave blank lines between examples to help readability. For example:

```Console
$ echo this is the first thing
this is the first thing

$ # This is some commentary on the next thing
$ echo another thing
another thing
```

Could be better as as an explicit testcase containing this kind of example.

Reply:
I just added a fixup commit that removes the blank lines stuff and adds this kind of more idiomatic example. I'll squash the commit if that looks good.

Reply:
I guess it's easier if I optimistically squash it. Let me know if there's anything else.